### PR TITLE
Task Detail Page

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
**Description of changes**
- Added Task Detail Page where user can delete, edit and mark complete.
- Added a logout button on the homepage.

**Associated ticket link**
https://trello.com/c/GcPL1b5J

**Screenshot / Screen Recording**

https://github.com/user-attachments/assets/f464d2ea-7b32-4fa7-97e0-06ee9f094084

### CODE REVIEW

File: modules/HomePage/components/TaskItem/TaskItem.tsx [hunk 1]
```diff
@@ -1,106 +1,56 @@
...
-import { FiEdit } from "react-icons/fi";
+import { Card, Group, Text, Badge } from "@mantine/core";
 
 import { useTasks } from "../TaskContext/TaskContext";
-import DeleteConfirmationModal from "../DeleteConfirmationModal/DeleteConfirmationModel";
-import ToDoForm from "../ToDoForm/ToDoForm";
 import { ITaskItemProps } from "./TaskItem.types";
-import { ITask } from "../TaskContext/TaskContext.types";
+import Link from "next/link";
+import { getPriorityColor } from "@/utils/priorityUtils";
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/components/TaskItem/TaskItem.tsx:7]: library and @ imports should be separate and at the top

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comment

**Review by ragib-hossain-sazim — CHANGES_REQUESTED** [review]
Logout functionality is not in the scope of this PR. Please create a separate ticket and PR to address this issue

**Review by ragib-hossain-sazim — APPROVED** [review]

**Review by ragib-hossain-sazim — CHANGES_REQUESTED** [review]
Why do we have have 14 commits? Also there seems to be some older commits in the PR.

**Comment by akib-raihan-sazim** [comment]
> Why do we have have 14 commits? Also there seems to be some older commits in the PR.

This happened while I was rebasing with the main branch. The older commits of main branch got appended.

**Comment by ragib-hossain-sazim** [comment]
@akib-raihan-sazim Please resolve the commit issue before merging it to main.

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**